src/main/java/Main.java

```diff
@@ -1,205 +1,148 @@
-     This method returns represents client calling with page size
-     */
-    private void QueryPageByPage() throws JSONException {
-        int pageSize = 500; //No of docs per page
-        int currentPageNumber = 1;
-        int documentNumber = 1;
-        String continuationToken = null;
-
-        do {
-            System.out.println("Page " + currentPageNumber);
-            String nextContinuationKey = "";
-
-            // Loads ALL documents for the current page
-            HashMap<String, List<Document>> docsForCurrPage = QueryDocumentsByPage(currentPageNumber, pageSize, continuationToken);
-
-            for (Map.Entry<String, List<Document>> entry : docsForCurrPage.entrySet()) {
-                System.out.println("Continuationkey" + entry.getKey() + " = " + entry.getValue());
-                nextContinuationKey = entry.getKey();
-                documentNumber++;
-            }
-
-            // Ensure the continuation token is kept for the next page query execution
-            continuationToken = nextContinuationKey;
-            System.out.println("continuationToken3 " + continuationToken);
-            currentPageNumber++;
-        } while (continuationToken != null);
+        //Demo Querying a Document with a list saved into Cache which can be used sliced in UI code for pagination
+        executeSimpleQueryWithList();
     }
 
-
-    /* This method returns continuation token with list of records for pagination on UI or REST API call*/
-    private HashMap<String, List<Document>> QueryDocumentsByPage(int currentPageNumber, int pageSize, String continuationToken) throws JSONException {
-
-        HashMap<String, List<Document>> map = new HashMap<>();
-
-        FeedOptions queryOptions = new FeedOptions();
-
-        // note that setMaxItemCount sets the number of items to return in a single page result
-        queryOptions.setMaxItemCount(pageSize);
-        queryOptions.setEnableCrossPartitionQuery(true);
-        queryOptions.setRequestContinuation(continuationToken);
+    private void queryDocumentsByPage() throws JSONException {
+        CosmosQueryRequestOptions queryOptions = new CosmosQueryRequestOptions();
 
         String collectionName = "volcanoCollection";
-        String sql = "SELECT * FROM volcanoCollection";
-
-        String collectionLink = String.format("/dbs/%s/colls/%s", databaseName, collectionName);
-        Observable<FeedResponse<Document>> queryObservable =
-                client.queryDocuments(collectionLink,
-                        sql, queryOptions);
+        String contanierSql = String.format("SELECT * from c where c.id = '%s'", collectionName);
+        System.out.println(contanierSql);
+        String itemSql = String.format("SELECT * FROM %s", collectionName);
+        System.out.println(itemSql);
+        CosmosPagedIterable<CosmosContainerProperties> queryObservable = database.queryContainers(contanierSql, queryOptions);
```

Comment:
This entire iterable logic can be replaced.

CosmosPagedIterable<CosmosContainerProperties> results = database.queryContainers(containerSql, queryOptions);
```java
// Iterating through all the results.
// If there are more than a single page of results, it will auto-magically fetch the next page.
for (CosmosContainerProperties cosmosContainerProperties : results) {
    String id = cosmosContainerProperties.getId();
    CosmosContainer container = database.getContainer(id);
    container.queryItems(itemSql, queryOptions, InternalObjectNode.class).forEach(item -> {
        System.out.println(item.toJson());
    });
    System.out.println(id);
}
```

Reply:
I haven't invested enough time to understand what items we are querying for in the container, but this usage of InternalObjectNode shouldn't be something we showcase.